Fix #16022: Auto-reject obsolete translation suggestions

sagangwee opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #16022.
2. This PR does the following: Auto-rejects obsolete translation suggestions when their corresponding exploration content or state gets deleted.

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes
made in this PR work correctly. If this PR is for a developer-facing feature,
provide the videos/screenshots of developer-facing interface.

Please also include videos/screenshots of the developer tools
browser console, so that we can be sure that there are no console errors.

If the changes in your PRs are autogenerated via a script and you cannot provide proof
for the changes then please leave a comment "No proof of changes needed because {{Reason}}"
and remove all the sections below.
-->
https://user-images.githubusercontent.com/9004726/220782843-4e1a644c-e864-427d-9344-a255810133cd.mov


Job screencast:

https://user-images.githubusercontent.com/9004726/220775627-0a33bc90-4c6a-4d77-9034-6605f5d4f0c0.mov


## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

sagangwee committed:
commit

sagangwee committed:
Merge remote-tracking branch 'upstream/develop' into develop

sagangwee committed:
Edit start script to use Python 3.8+

sagangwee committed:
Merge remote-tracking branch 'upstream/develop' into develop

sagangwee committed:
Merge remote-tracking branch 'upstream/develop' into develop

sagangwee committed:
Initial commit

sagangwee committed:
Update comment

sagangwee committed:
Merge conflicts

sagangwee committed:
Remove python-version

sagangwee commented:
@seanlip @vojtechjelinek @aks681 PTAL, thanks!

vojtechjelinek commented on .python-version:
```diff
@@ -0,0 +1 @@
+oppia_3.8_x86
```
Unexpected change? I think you can maybe add .python-version to .gitignore, if you want to use it.

sagangwee replied:
Ah, thanks. Forgot to push the change removing this.

vojtechjelinek commented on core/storage/suggestion/gae_models.py:
```diff
@@ -726,6 +726,27 @@ def get_in_review_translation_suggestions_with_exp_ids_by_offset(
+            list(SuggestionModel). A list of suggestions matching the supplied
+            exp_id.
+        """
+        return cls.get_all().filter(datastore_services.all_of(
+            cls.status == STATUS_IN_REVIEW,
+            cls.suggestion_type == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+            cls.target_id == exp_id
+        )).fetch(feconf.DEFAULT_SUGGESTION_QUERY_LIMIT)
```
Did you locally test the code that calls this function? It should generate a new entry in _index.yaml_ I think.

sagangwee replied:
Hmm, I did test this locally, but there were no `index.yaml` changes. Historically, I've had this issue, so it may be something to do with my personal setup.

seanlip replied:
Hmm, you might need to update index.yaml manually then... 

@vojtechjelinek is there a way for e.g. someone else to pull the code, run some script, and generate the necessary changes that should be made? We might want to document that (and/or maybe have some sort of standard diagnostic test that devs can run to see whether this works on their machine -- it could work by asking them to add a function with a query, running the script, and seeing if index.yaml changes).

sagangwee replied:
Updated the index manually for now.

vojtechjelinek replied:
@seanlip It is pretty much what you said, pull, run and see if the index.yaml is updated. Not sure where or how to document this.

seanlip replied:
@vojtechjelinek I would put it in a wiki page titled "Things to keep in mind when making storage layer changes" or similar. Filed https://github.com/oppia/oppia-web-developer-docs/issues/122 to track.

vojtechjelinek commented on core/domain/suggestion_services.py:
```diff
@@ -902,6 +902,54 @@ def auto_reject_translation_suggestions_for_exp_ids(exp_ids: List[str]) -> None:
+def auto_reject_translation_suggestions_for_content_ids(
+    exp_id: str,
+    state_name: str,
+    content_ids: Set
```
What kind of set? `Set[str]` I assume?

sagangwee replied:
Right. Updated.

vojtechjelinek commented on core/domain/suggestion_services.py:
```diff
@@ -902,6 +902,54 @@ def auto_reject_translation_suggestions_for_exp_ids(exp_ids: List[str]) -> None:
+def auto_reject_translation_suggestions_for_state(
+    exp_id: str,
+    state_name: str
+) -> None:
```
```suggestion
    exp_id: str, state_name: str
) -> None:
```

sagangwee replied:
Done.

vojtechjelinek commented on core/domain/exp_services_test.py:
```diff
@@ -3846,6 +3913,73 @@ def test_update_interaction_customization_args(self) -> None:
+                            'unicode_str': 'Continue'
+                        }
+                    }
+                }),
```
Can you initialize the full change list here and then just pass it? It is a bit hard to read now.

sagangwee replied:
Done.

vojtechjelinek commented on core/domain/exp_services_test.py:
```diff
@@ -3607,6 +3610,70 @@ def test_delete_state_cmd(self) -> None:
+                }
+            })
+        ]
+        exp_services.update_exploration(
+            self.owner_id, self.EXP_0_ID, change_list, '')
```
Can you just provide some commit message instead of `''` (for better readability).

vojtechjelinek replied:
ditto below

sagangwee replied:
Sure done.

vojtechjelinek reviewed (changes requested):
Thanks! Took a pass.

sagangwee commented:
Currently updating to account for https://github.com/oppia/oppia/pull/17200 changes.

sagangwee committed:
Address reviewer comments

sagangwee committed:
Revert core/domain/translation_fetchers.py

sagangwee reviewed (commented):
@vojtechjelinek Responded to your comments, thanks!

seanlip commented on core/domain/exp_services_test.py:
```diff
@@ -3607,6 +3610,92 @@ def test_delete_state_cmd(self) -> None:
+            exploration.next_content_id_index
+        )
+        content_id = content_id_generator.generate(
+                    translation_domain.ContentType.CONTENT)
```
Should only be indented by 4 from the previous line.

sagangwee replied:
Ah, thanks. Done.

sagangwee replied:
Ah, thanks. Done.

seanlip reviewed (changes requested):
Thanks @sagangwee! No particular concerns code-wise, but could you please upload proof that the changes are correct (per the guidance in the PR description) that shows obsolete suggestions are rejected and non-obsolete suggestions are not? Thanks!

EDIT: Also, would like to see what the submitter sees on their "history dashboard" after the suggestion is rejected.

sagangwee committed:
Remove obsolete check from suggestion.py

sagangwee commented:
@seanlip Ah, thanks. Added a screencast and addressed your comment.

sagangwee reviewed (commented):
Woops. Adding back gitignore and setup.py...

sagangwee committed:
Add back changes

sagangwee committed:
Adjust comment

sagangwee committed:
Update yaml

sagangwee committed:
Move manual index

sagangwee commented:
Currently updating jobs in https://github.com/oppia/oppia/blob/develop/core/jobs/batch_jobs/rejecting_suggestion_for_invalid_content_ids_jobs.py to account for the translation structure changes of https://github.com/oppia/oppia/pull/17200. Will file job request once that's ready.

vojtechjelinek commented on core/controllers/suggestion.py:
```diff
@@ -1186,8 +1183,14 @@ def _construct_exploration_suggestions(
         exploration = exp_id_to_exp[suggestion.target_id]
-        content_html = exploration.get_content_html(
-            suggestion.change.state_name, suggestion.change.content_id)
+        content_html = ''
```
Is this even needed?

sagangwee replied:
I don't believe so in Python, but I put it here since I'm only catching one type of exception so to account for some other type of error.

vojtechjelinek commented on core/domain/suggestion_services.py:
```diff
@@ -902,6 +902,29 @@ def auto_reject_translation_suggestions_for_exp_ids(exp_ids: List[str]) -> None:
         suggestion_models.INVALID_STORY_REJECT_TRANSLATION_SUGGESTIONS_MSG)
 
 
+def auto_reject_translation_suggestions_for_content_ids(
```
Can we rather name this 
```suggestion
def reject_translation_suggestions_for_content_ids(
```

I don't see why we need the `auto` we still have to run this manually. This might be also updated for the other functions like this.

sagangwee replied:
Future lesson content changes that make translation suggestions obsolete will automatically trigger rejections. I think the "auto" part is mostly to signal that the suggestion reviewer is set to OppiaSuggestionBot and to differentiate it from a normal rejection from a human user.

vojtechjelinek replied:
Hmm, the "auto" prefix doesn't do that for me, but on the other hand I do not have any suggestion for a different naming.

sagangwee replied:
Ack.

vojtechjelinek commented on index.yaml:
```diff
@@ -1,5 +1,11 @@
 
+- kind: GeneralSuggestionModel
+  properties:
+  - name: status
+  - name: suggestion_type
+  - name: target_id
```
```suggestion
  - name: deleted
  - name: status
  - name: suggestion_type
  - name: target_id
```

vojtechjelinek replied:
Also, put this below the comment below.

sagangwee replied:
Added the deleted attribute, but for placement, I was going off of "If you want to manage some indexes manually, move them above the marker line." . Is that not right?

vojtechjelinek replied:
No, it should be generated automatically, but for some reason it does not work for you, but we still should put it below.

sagangwee replied:
Done.

vojtechjelinek reviewed (changes requested):
THanks! Took a pass.

seanlip commented on core/controllers/suggestion.py:
```diff
@@ -1186,8 +1183,14 @@ def _construct_exploration_suggestions(
+        except ValueError:
+            # If the exploration content is no longer available, return the
+            # suggestion's content.
+            content_html = suggestion.change.content_html
```
Conceptually, isn't this weird -- shouldn't it be the empty string or None to denote that the content in the original exploration no longer exists?

sagangwee replied:
That's a good point. I'll set it to an empty string here, and add behavior in the frontend.

seanlip reviewed (changes requested):
Thanks @sagangwee, just left one inline note.

Also one thing strikes me as weird -- I'm kind of confused by the modal that pops up at the end of the video. If this is for folks to see their previous contributions then why is the popup called "Review translation contributions"  and why is the note suggesting to edit the translation as needed? This seems like it doesn't make sense from the perspective of a user looking at their past submissions (presumably in view-only mode).

sagangwee committed:
beam job

sagangwee committed:
update index

sagangwee committed:
linting

sagangwee commented:
@seanlip Tweaked the modal a bit to make it less confusing for non-reviewers. Thanks!

@vojtechjelinek Responded to your comments. Thanks! Also added the necessary beam job.

sagangwee commented:
@aks681 PTAL for codeowners, thanks!

seanlip reviewed (changes requested):
Thanks @sagangwee! Took a look, couple of comments:

- Did you miss this comment? https://github.com/oppia/oppia/pull/17259#discussion_r1101790948 I don't think it's been responded to...
- The new modal looks good, but I have a further question about the UI. If the suggestion is obsolete, then why does it show up in the submitter's dashboard as "revisions requested"?

vojtechjelinek commented:
@sagangwee Some of my comments are not replied to.

sagangwee committed:
obsolete label

sagangwee reviewed (commented):
@seanlip @vojtechjelinek Ah, sorry. GitHub didn't push my "in-review" comments. 

Sean, I also updated the "Revisions Requested" label for the obsolete case:

<img width="1449" alt="Screenshot 2023-02-24 at 3 33 20 PM" src="https://user-images.githubusercontent.com/9004726/221285955-8fcc7a1d-a5d1-46d2-8b58-d8ea9808f04f.png">

sagangwee commented:
@kevintab95 PTAL, thanks!

seanlip commented on core/controllers/suggestion.py:
```diff
@@ -1186,8 +1183,13 @@ def _construct_exploration_suggestions(
         exploration = exp_id_to_exp[suggestion.target_id]
-        content_html = exploration.get_content_html(
-            suggestion.change.state_name, suggestion.change.content_id)
+        content_html: Union[str, List[str]] = ''
```
@sagangwee Can we set this to null instead? That is semantically more correct, otherwise it suggests that the content HTML is the empty string (which is a valid value).

sagangwee replied:
Done.

seanlip reviewed (changes requested):
Thanks @sagangwee! Just one question.

vojtechjelinek commented on core/jobs/batch_jobs/rejecting_suggestion_for_invalid_content_ids_jobs.py:
```diff
@@ -45,273 +43,144 @@
+                    # Exploration ID.
+                    str,
+                    # Obsolete content dict.
+                    List[Dict[str, str]]]]]) = []
```
```suggestion
        obsolete_translation_suggestion_error_report: List[
            Dict[str, Union[
                # Exploration ID.
                str,
                # Obsolete content dict.
                List[Dict[str, str]]
            ]]
        ] = []
```

vojtechjelinek reviewed (changes requested):
Thanks! Took a pass.

sagangwee committed:
Update core/jobs/batch_jobs/rejecting_suggestion_for_invalid_content_ids_jobs.py

Co-authored-by: Vojtěch Jelínek <vojtech.jelinek@hey.com>

sagangwee committed:
Review comments

sagangwee commented:
@seanlip @vojtechjelinek Responded to your comments, thanks!

sagangwee commented:
@aks681 PTAL, thanks!

sagangwee commented:
@kevintab95 PTAL, thanks!

seanlip reviewed (approved):
Thanks @sagangwee! LGTM for codeowners.

sagangwee committed:
lamda

aks681 reviewed (approved):
Lgtm as codeowner

sagangwee committed:
TypeScript compilation

sagangwee committed:
code coverage

sagangwee committed:
Merge remote-tracking branch 'upstream/develop' into auto-reject-obsolete

sagangwee committed:
coverage

vojtechjelinek commented:
@sagangwee Please request a backup server testing for the new job.

sagangwee committed:
beam job coverage

sagangwee commented:
@vojtechjelinek Submitted a testing request, thanks!

sagangwee committed:
suggestion.py coverage

sagangwee committed:
Merge branch 'develop' into auto-reject-obsolete

won369369 committed:
Fix part of #14219: Increase Per-File Backend Test Coverage for visualization_registry (#17412)

removed visualization_registry_test from scripts/backend_tests_incomplete_coverage.txt

Jaggenaut committed:
Fix issue #17498 Visibility of dropdown icon on click and added focus on user Account section. (#17513)

Fixed hover problem of downarrow plus added the focus on user account section

SD-13 committed:
Fix #16368: Lint check for *_jobs in registry (#17488)

* add lint check for *_jobs

Commit:
Update version to 3.3.0 (#17563)

* Update core/feconf.py

* Update package.json

---------

Co-authored-by: github-actions[bot] <github-actions@noreply.com>

kashishbhandula committed:
Console error in the story preview screen #10610 (#17476)

small changes

walnuthanhan committed:
Fix #6193: Center CORRECT! Message Box in Exploration (#17550)

Fix display of CORRECT! message

Co-authored-by: Han Trinh <tringiahan1308@gmail.com>

prafulbbandre committed:
Fix part of #17415:Overlaying of Next lesson button on next page button cards solved. (#17467)

* overlay solved

* Update conversation-skin.component.css

* Update conversation-skin.component.css

* Update conversation-skin.component.css

* Update conversation-skin.component.css

* Update conversation-skin.component.css

* Update top-navigation-bar.component.css

* Update conversation-skin.component.css

* Update top-navigation-bar.component.css

* Update progress-nav.component.html

* Update conversation-skin.component.css

* removed space

seanlip committed:
Update authors since d40bb8a (#17556)

Priyansh61 committed:
Fix part of #17553 Broken Toaster messages  (#17571)

added missing ngx-toaster.css

vojtechjelinek reviewed (changes requested):
Thanks! Only note that I have after running it on the backup server, can you please output the total number of processed suggestions in `AuditTranslationSuggestionsWithMissingContentIdJob` and `RejectTranslationSuggestionsWithMissingContentIdJob` to be sure that we went through all the suggestions and to be able to run the `AuditTranslationSuggestionsWithMissingContentIdJob` at the end and see some output (because the other output will be resolved by the reject job).

kashishbhandula committed:
Unable to add other RTE tags inside tabs tag and collapsibles tag #15935 (#17489)

* added css property

* fixed lints

JeeveshGarg committed:
MS-last(Improve frontend types): Remove Unknown (#16597)

* ng-cli

* Successful build

* Angular v14 angular.json

* AOT working, old build not running

* Add webpack module loader

* Fix build after merge from develop

* Fix prod build

* Remove duplicate css and add css styleUrls

* Remove repeated assets.ts code file and fix more css files

* Fix ng build and add start commands

* Add support for production

* Fix logic

* Add more fixes to wrong html files

* Add a mock hashes file to satisfy angular compiler

* New fixes for the template errors

* Bug fixes

* Fix new changes in develop

* Fix new changes in develop

* Review changes

* Fix be tests

* Fix mypy types

* Fix errors

* Lint and backend test fixes

* Fix review comments and add tests

* Merge from upstream/develop

* Fix unit tests

* Fix lint errors

* Fix tests

* Fix tests

* Add a deafult lang direction when no lang dir is present in the url

* Fix build errors in latest develop

* Move document attribution lang setting logic outside the if case

* Fix backend tests

* Upgrade angular and ts version

* yarn lock changes

* Fix backend tests

* Partially finished backend tests

* Downgrade angular back to 11 and move refactored code to ignore list in ts checks

* Fix develop

* Add files to deny lists

* Fix tests

* Fix tests again

* Fix lint

* Fix lint errors

* Fix backend tests

* unknown remove

* unknown remove

* nit

* nit

* nit

* lint

* Fix errors in ng build prod

* Review comments

* ts-solve

* nit

* nit

* nit

* nit

* nit

* nit

* nit

* nit

* frontend solve

* frontend solve

* mypy solve

* lint

* lint

* lint

* comments

* comments

* comments

* try

* lint solve

* All TODO removed

* Backend coverage left only

* whole unknown good to go now

* front test coner

* some lint

* lint

* backend

* Backend test

* Backend test

* Fix upstream/develop

* Addressing review comments

* Fix typings

* Add dist to copied files in e2e tests

* Fix template urls

* Fix multiple browser module imports

* Fix backend tests

* Fix backend tests

* Fix backemd tests

* Fix comments

* push aot typing

* srijan advice

* nit

* lint

* try

* ft solve

* lint

* lighthouse solve

* linter done

* comments

* ts

* srijan comments

* done

* done

* done

* changes

* lint

* comments by chris

* remove lint check

* comment

* comment

* sean comment

* vojta comment

* vojta comment

* done

* done

* done

---------

Co-authored-by: Vasa Srijan Reddy <redvasa@amazon.com>
Co-authored-by: Srijan Reddy <srijan@srijanreddy.com>
Co-authored-by: Vojtěch Jelínek <vojtin.j@gmail.com>
Co-authored-by: Vojtěch Jelínek <vojtech.jelinek@hey.com>

Kej-r03 committed:
Fix improvementsTabIsEnabled console error (#17450)

Co-authored-by: Priyansh Mehta <96513964+Priyansh61@users.noreply.github.com>

SubhashThenua committed:
Fix part of #17553:  Promo bar is completely readable  (#17573)

* Promo-bar regrain to original UI

* Promo UI fix

* linter-fix

---------

Co-authored-by: JSRKJSRK <SubhashThenua>

sagangwee committed:
report total count

sagangwee committed:
Merge branch 'develop' into auto-reject-obsolete

sagangwee commented:
@vojtechjelinek Thanks. Added the total processed count. Also, the backend test coverage check keeps failing, but I can't figure out which lines of code it's complaining about. I checked the coverage report, and none of the lines pertain to code I've changed in this PR. Any tips? I've already looked at https://github.com/oppia/oppia/wiki/Backend-tests#coverage-reports.

vojtechjelinek commented:
@sagangwee 
![image](https://user-images.githubusercontent.com/10142938/223229487-86a283f9-9c9c-420e-8539-eee61bbe09eb.png)

Which means 4802-4805 lines in state_domain.py are not covered.

vojtechjelinek commented on core/feconf.py:
```diff
@@ -1624,6 +1624,11 @@ def get_empty_ratings() -> Dict[str, int]:
 
+# Auto-generated review message for an obsolete translation suggestion, i.e. a
+# suggestion corresponding to content that no longer exists.
+OBSOLETE_TRANSLATION_SUGGESTION_REVIEW_MSG = (
+    constants.OBSOLETE_TRANSLATION_SUGGESTION_REVIEW_MSG)
```
Why is this needed why you cannot use `constants.OBSOLETE_TRANSLATION_SUGGESTION_REVIEW_MSG` directly.

sagangwee replied:
Ah, thanks. Didn't realize you could access `constants` in backend files.

vojtechjelinek commented on core/jobs/batch_jobs/rejecting_suggestion_for_invalid_content_ids_jobs.py:
```diff
@@ -45,273 +43,174 @@
+        suggestion_dicts = _get_suggestion_dicts(self.pipeline)
+        total_processed_suggestions_count_job_run_results = (
+            suggestion_dicts
+            | 'Get suggestions' >> beam.Map(
+                lambda suggestions_dict: suggestions_dict['suggestions']
+                )
+            | 'Flatten suggestions' >> beam.FlatMap(lambda x: x)
```
```suggestion
            | 'Get suggestions' >> beam.FlatMap(
                lambda suggestions_dict: suggestions_dict['suggestions']
            )
```

vojtechjelinek replied:
ditto elsewhere

sagangwee replied:
Done. Thanks!

vojtechjelinek reviewed (changes requested):
Thanks! Left a few other comments. Please request testing again as the PR changed a bit with the merged changes. Thanks!

sagangwee committed:
address review comments

sagangwee committed:
Merge branch 'auto-reject-obsolete' of github.com:sagangwee/oppia into auto-reject-obsolete

sagangwee commented:
@vojtechjelinek Thanks, I submitted a new request (should be the same as the old one) and addressed your comments.

vojtechjelinek commented:
@sagangwee I don't see any new request.

sagangwee commented:
@vojtechjelinek Submitted a new request from a blank form, thanks.

vojtechjelinek commented:
@sagangwee Thanks! Hey, @kevintab95 or @DubeySandeep can you please run the job again? I won't be able to look into it as I'm finishing other stuff.

sagangwee committed:
Merge branch 'develop' into auto-reject-obsolete

sagangwee commented:
@kevintab95 or @DubeySandeep friendly ping.

cc @vojtechjelinek

DubeySandeep commented:
@sagangwee I'll run the job tomorrow and update you with the result!

DubeySandeep commented:
@sagangwee Can you please reply to the comments in the doc? (I'm going to run the audit job meanwhile.)

DubeySandeep commented:
@sagangwee I've deployed your commits on the backup server but I'm unable to find the required jobs on the release-coordinator's page. I think you have to import the new job's file in the registry (not sure!)

Also, please make sure to test the job locally through [the release coordinator page](https://github.com/oppia/oppia/wiki/Apache-Beam-Jobs#running-apache-beam-jobs).

DubeySandeep commented:
@sagangwee Looks like the jobs file is already imported in the registry, not sure why I'm unable to access it in the server. I'm looking into it!

DubeySandeep commented:
@sagangwee I'm able to run the jobs now. I think earlier it was not available on the jobs page as the server traffic migration was in the process and took mins to complete (internally). I've reported this to the release team and requested them to find a way to check the status of the traffic migration to avoid such issues in the future (sorry for the trouble!)

DubeySandeep commented:
@sagangwee I have run the audit job on the backup server and looks like we don't have any such translation suggestions.

@vojtechjelinek Considering the job is already ran on the backup server all the required models are already migrated. We have to reset the server data to re-test the job, I'm not sure how significant the changes are in the jobs, do you think we should reset the server and rerun the jobs?

sagangwee commented:
@DubeySandeep I've tested the job locally as seen in the PR description. I also responded to your comments in the testing doc. Thanks!

sagangwee committed:
Merge branch 'develop' into auto-reject-obsolete

sagangwee commented:
@DubeySandeep @vojtechjelinek Any update here?

DubeySandeep commented:
@sagangwee Looks like @vojtechjelinek is on vacation, we have initiated a reset on backup servers which will take 6 hrs. and I'll make sure to re-run the jobs tonight!

sagangwee commented:
@DubeySandeep Ah, thanks!

DubeySandeep commented:
@sagangwee I've shared the audit job result with you via email, I'll run the other job once you'll reply to the email.

sagangwee commented:
@DubeySandeep The audit job result LGTM, thanks!

sagangwee committed:
Merge branch 'develop' into auto-reject-obsolete

DubeySandeep reviewed (approved):
@sagangwee The job ran successfully on the backup server, I've shared the logs through email. Feel free to merge your PR!

vojtechjelinek reviewed (approved):
LGTM!

sagangwee committed:
Merge remote-tracking branch 'upstream/develop' into auto-reject-obsolete

sagangwee committed:
Merge branch 'auto-reject-obsolete' of github.com:sagangwee/oppia into auto-reject-obsolete

sagangwee committed:
audit suggestion count
